src/hooks/Circles/utils/optimisticallyDeletePosts.ts

```diff
@@ -54,10 +55,28 @@ const removeEdgeById = (currentData: InfinitePostsData, id: string) =>
 const removeRepliesById = (
   currentData: PostDetailsPostQueryResponse,
   id: string,
+  depth: number = 7, // set some default max depth
 ) => {
-  const deleted = remove(
-    currentData?.post.replies.edges,
-    (edge) => edge.node.id === id,
-  );
-  currentData.post.replyCount -= deleted.length;
+  let count = 0;
+  const recursiveRemove = (post: Post, currentDepth: number) => {
+    if (currentDepth > depth) {
+      return;
+    }
+
+    const deleted = remove(post.replies.edges, (edge) => edge.node.id === id);
+    post.replyCount -= deleted.length;
+
+    // Recurse into the replies of the remaining edges
+    post.replies.edges.forEach((edge) => {
+      if (edge.node.replies && edge.node.replies.edges) {
+        recursiveRemove(edge.node, currentDepth + 1);
+      }
+    });
```

Comment:
In all for loops / recursion in this PR (including this one), there aren't any network requests inside of these, right?  If so, I think we should definitely be doing this on the backend instead.  Even if not, It feels like the client is having to work pretty hard to keep all of this straight - is there any opportunity (now or down the road) to make this client-side logic simpler (or removed altogether)?

Reply:
This is purely to update the local cache when the user performs a post mutation, i.e., update a post message, delete a post, add a new reaction. This seemingly expensive logic shouldn't be _that_ heavy given it's only iterating a fairly small dataset (whatever is available in the cache locally) and only occurs when such a mutation is issued. That said we could probably make some gains by using store indexes to access the data more directly without having to search for it by id each time.

Reply:
Another optimization is that I don't think we need to recur if `deleted.length > 0` an reply should only exist at one point in the post/reply tree

Reply:
Handled by the refactor